internal/server/deviceserver.go

```diff
@@ -37,3 +40,16 @@ func (s *deviceServer) Send(ctx context.Context, d *pb.Data) (*pb.Receipt, error
 	// Respond to the start request that the work was accepted.
 	return &pb.Receipt{}, nil
 }
+
+// Disconnect implements the "Disconnect" method of the Worker gRPC service.
+func (s *deviceServer) Disconnect(ctx context.Context, in *pb.Empty) (*pb.DisconnectResponse, error) {
+	log.Info("received worker disconnect request")
```

Comment:
Will we accept new workload requests at this point? I think once we receive disconnect we should not accept any new workloads.

Reply:
It means I need to add a lock for that section and also for the creating